sources/api/migration/migration-helpers/src/common_migrations.rs

```diff
@@ -1260,6 +1260,136 @@ mod test_add_metadata {
+                    setting_metadata.setting
+                );
+            }
+        }
+        Ok(input)
+    }
+
+    fn backward(&mut self, input: MigrationData) -> Result<MigrationData> {
+        println!(
+            "RemoveMetadataMigration({:?}) has no work to do on downgrade.",
+            &self.0
+        );
+        Ok(input)
+    }
+}
+
+#[cfg(test)]
+mod test_remove_metadata {
+    use super::{AddMetadataMigration, RemoveMetadataMigration, SettingMetadata};
+    use crate::{Migration, MigrationData};
+    use maplit::hashmap;
+    use std::collections::HashMap;
+
+    #[test]
+    fn forward() {
+        let data = MigrationData {
+            data: HashMap::new(),
+            metadata: hashmap! {
+                "hi".into() => hashmap!{"there".into() => "whatever".into() },
+            },
+        };
+        let result = RemoveMetadataMigration(&[SettingMetadata {
+            setting: "hi",
+            metadata: &["there"],
+        }])
+        .forward(data)
+        .unwrap();
+        assert_eq!(result.metadata, hashmap! { "hi".into() => HashMap::new() });
+    }
+
+    #[test]
+    fn forward_noop() {
+        let data = MigrationData {
+            data: HashMap::new(),
+            metadata: hashmap! {
+                "hi".into() => hashmap!{"there".into() => "whatever".into() },
+            },
+        };
+        let result = RemoveMetadataMigration(&[SettingMetadata {
+            setting: "hi",
+            metadata: &["which"],
+        }])
+        .forward(data)
+        .unwrap();
+        assert_eq!(
+            result.metadata,
+            hashmap! { "hi".into() => hashmap!{"there".into() => "whatever".into() } }
+        );
+    }
+
+    #[test]
+    fn backward() {
+        let data = MigrationData {
+            data: HashMap::new(),
+            metadata: hashmap! {
+                "hi".into() => hashmap!{"there".into() => "whatever".into()},
+            },
+        };
+        let result = RemoveMetadataMigration(&[SettingMetadata {
+            setting: "hi",
+            metadata: &["there"],
+        }])
+        .backward(data)
+        .unwrap();
+        assert_eq!(
+            result.metadata,
+            hashmap! {
+                "hi".into() => hashmap!{"there".into() => "whatever".into()},
+            }
+        );
+    }
+
+    #[test]
+    fn backward_noop() {
+        let data = MigrationData {
+            data: HashMap::new(),
+            metadata: hashmap! {
+                "hi".into() => hashmap!{"sup".into() => "wassup".into()},
+            },
+        };
+        let result = RemoveMetadataMigration(&[SettingMetadata {
+            setting: "hi",
+            metadata: &["there"],
+        }])
+        .backward(data)
+        .unwrap();
+        assert_eq!(
+            result.metadata,
+            hashmap! {
+                "hi".into() => hashmap!{"sup".into() => "wassup".into()},
+            }
+        );
+    }
+}
```

Comment:
I don't think you need this test unless I'm missing something subtle, since backwards is always a no-op.

Reply:
> I don't think you need this test unless I'm missing something subtle, since backwards is always a no-op.

Since this migration is removing something, backwards should put it back. Right?

Reply:
> Since this migration is removing something, backwards should put it back. Right?

`storewolf` will put it back

Reply:
Yeah I validated in my testing that storewolf indeed put them back